pkg/provisioningv2/rke2/planner/encryptionkeyrotation.go

```diff
@@ -31,52 +81,46 @@ const (
 // match the init node, and either finishing if the status is reencrypt_finished, or applying the next secrets-encrypt
 // command.
 func (p *Planner) rotateEncryptionKeys(cp *rkev1.RKEControlPlane, clusterPlan *plan.Plan) error {
+	// ensure operating on latest control plane object
+	cp, err := p.rkeControlPlanes.Get(cp.Namespace, cp.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
 	if !shouldRotateEncryptionKeys(cp) {
 		return nil
 	}
 
 	if shouldRestartEncryptionKeyRotation(cp) {
 		return p.restartEncryptionKeyRotation(cp)
 	}
-	_, _, leader, err := p.findInitNode(cp, clusterPlan)
+
+	leader, err := p.encryptionKeyRotationElectLeader(cp, clusterPlan)
 	if err != nil {
 		return err
 	}
-	if leader == nil {
-		logrus.Error("Could not rotate encryption keys: unable to find init node")
-		return nil
-	}
+
+	logrus.Debugf("rkecluster %s/%s: current encryption key rotation phase: [%s]", cp.Namespace, cp.Spec.ClusterName, cp.Status.RotateEncryptionKeysPhase)
 
 	switch cp.Status.RotateEncryptionKeysPhase {
-	case rkev1.RotateEncryptionKeysPhaseStart:
-		return p.encryptionKeyRotationStart(cp, leader)
-	case rkev1.RotateEncryptionKeysPhaseRestartLeader:
-		return p.encryptionKeyRotationRestartLeader(cp, leader)
-	case rkev1.RotateEncryptionKeysPhaseVerifyLeaderStatus:
-		return p.encryptionKeyRotationVerifyLeaderStatus(cp, leader)
-	case rkev1.RotateEncryptionKeysPhaseRestartFollowers:
-		return p.encryptionKeyRotationRestartFollowers(cp, clusterPlan, leader)
+	case rkev1.RotateEncryptionKeysPhaseRestartNodes:
+		return p.encryptionKeyRotationRestartNodes(cp, clusterPlan, leader)
 	case rkev1.RotateEncryptionKeysPhaseApplyLeader:
-		return p.encryptionKeyRotationApplyLeader(cp, leader)
+		return p.encryptionKeyRotationApplyLeader(cp, clusterPlan, leader)
 	}
 
 	return p.enqueueAndSkip(cp)
 }
 
-// shouldRotateEncryptionKeys `true` if encryption key rotation has been requested or has not finished
+// shouldRotateEncryptionKeys `true` if encryption key rotation has been requested or has not finished.
 func shouldRotateEncryptionKeys(cp *rkev1.RKEControlPlane) bool {
-	// The controlplane must be initialized before we rotate anything
-	if !cp.Status.Initialized {
-		return false
-	}
-	// if a spec is not defined there is nothing to do
-	if cp.Spec.RotateEncryptionKeys == nil {
-		return false
+	// Do not stop rotating for any reason, except when encryption key rotation cannot be completed
+	if rke2.Ready.IsUnknown(cp) && rke2.Ready.GetReason(cp) == rke2.RotatingEncryptionKeysReason {
```

Comment:
If any function during the encryption process returns an error, even an `ErrWaiting`, then this won't be true and, if I understand this code correctly, encryption key rotation will stop.

For example, if an error is returned, then the `Ready` condition will be `false`. If an `ErrWaiting` is returned (like may happen below), then the reason on the `Ready` condition will be `Waiting`.

I know you tested this, but I am concerned that if a error happens for some reason then this whole process might stall.